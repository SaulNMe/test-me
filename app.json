{
  "expo": {
    "name": "Qualificame",
    "slug": "qualificame-native",
    "privacy": "public",
    "sdkVersion": "33.0.0",
    "platforms": [
      "ios",
      "android"
    ],
    "version": "1.0.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splashCustom.png",
      "resizeMode": "cover",
      "backgroundColor": "#ffffff",

    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true
    }
  }
}
