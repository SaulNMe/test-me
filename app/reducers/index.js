import Events, * as fromEvents from 'qualificame-native/app/reducers/EventsReducer.js';
import Auth, * as fromAuth from 'qualificame-native/app/reducers/AuthReducer.js';
import TokenRefresh, * as fromTokenRefresh from 'qualificame-native/app/reducers/TokenRefreshReducer.js'
import Kiosks, * as fromKiosks from 'qualificame-native/app/reducers/KiosksReducer.js';
import Reports, * as fromReports from 'qualificame-native/app/reducers/ReportsReducer.js';
import Choices, * as fromChoices from 'qualificame-native/app/reducers/ChoicesReducer.js';
import Alerts, * as fromAlerts from 'qualificame-native/app/reducers/AlertsReducer.js';

import { combineReducers } from 'redux'

const rootReducer = combineReducers({
    Auth,
    TokenRefresh,
    Kiosks,
    Reports,
    Events,
    Choices,
    Alerts,
});

export default rootReducer;

//Auth
export const getIsLoggedIn = state =>
	fromAuth.getIsLoggedIn(state.Auth);

export const getIsLoading = state =>
	fromAuth.getIsLoading(state.Auth);

export const getAuthData = state =>
	fromAuth.getAuthData(state.Auth);

export const getUserData = state =>
	fromAuth.getUserData(state.Auth);

//Token Refresh
export const getIsRefreshingToken = state =>
	fromTokenRefresh.getIsRefreshingToken(state.TokenRefresh);

export const getRefreshCall = state =>
	fromTokenRefresh.getRefreshCall(state.TokenRefresh);

//Kiosks
export const getKiosks = state =>
	fromKiosks.getKiosks(state.Kiosks);

export const getKioskById = (state, kioskId) =>
	fromKiosks.getKioskById(state.Kiosks, kioskId)

export const getIsLoadingKiosks = state =>
	fromKiosks.getIsLoadingKiosks(state.Kiosks);

export const getErrorKiosks = state =>
	fromKiosks.getErrorKiosks(state.Kiosks);

//Reports

export const getReports = state =>
	fromReports.getReports(state.Reports);

export const getIsLoadingReports = state =>
	fromReports.getIsLoadingReports(state.Reports);
export const getErrorReports = state =>
	fromReports.getErrorReports(state.Reports);

// Events
export const getEvents = state =>
	fromEvents.getEvents(state.Events);

export const getEventsGroupedByDay = (state, ids) =>
	fromEvents.getEventsGroupedByDay(state.Events, ids);

export const getSelectedMonth = state => 
	fromEvents.getSelectedMonth(state.Events);

export const getEventsSelectedDateRange = state => 
	fromEvents.getEventsSelectedDateRange(state.Events);

// Choices
export const getChoices = state =>
	fromChoices.getChoiceById(state.Choices);

// Alerts 
export const getAlerts = state =>
	fromAlerts.getAlerts(state.Alerts);

export const unReadAlerts = state =>
	fromAlerts.unReadAlerts(state.Alerts);