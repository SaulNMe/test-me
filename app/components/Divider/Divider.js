import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { Colors  } from 'qualificame-native/app/styles';

import styles from './DividerStyle';

export default class Divider extends PureComponent {
  render() {
	return (
		<View style={[styles.divider, {backgroundColor: Colors[this.props.color]}]} />
	);
  }	
}

Divider.propTypes = {
	color: PropTypes.string
}

Divider.defaultProps = {
	color: 'gray'
}
