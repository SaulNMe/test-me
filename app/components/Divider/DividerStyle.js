import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	divider: {
		height: 1,
		color: Colors.light,
		width:'100%'
	}
});
