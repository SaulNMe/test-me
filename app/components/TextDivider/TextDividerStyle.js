import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-native/app/styles';

export default StyleSheet.create({
	...ApplicationStyles.flexBox,
	...ApplicationStyles.margins,
	divider: {
		width: '100%',
		borderBottomWidth: 1
	}
});
