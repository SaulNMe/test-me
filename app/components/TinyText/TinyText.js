import React, { PureComponent } from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';
import { Fonts, Colors } from 'qualificame-native/app/styles';
import styles from './TinyTextStyle';

export default class TinyText extends PureComponent {
	render() {
		const textColor = {color: Colors[this.props.color]};
		const weight = {fontWeight: Fonts.weight[this.props.weight]};
		const align = {textAlign: this.props.align};

		return (
			<React.Fragment>
				<Text style={[styles.text, textColor, weight, align]}>{this.props.text}</Text>
			</React.Fragment>
		);
	}
}

	TinyText.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		weight: PropTypes.string,
		align: PropTypes.string,
	}

	TinyText.defaultProps = {
		text: "",
		color: "dark",
		weight: "regular",
		align: "left"
	}
