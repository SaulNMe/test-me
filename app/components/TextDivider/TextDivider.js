import React, { PureComponent } from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import TinyText from 'qualificame-native/app/components/TinyText';
import Divider from 'qualificame-native/app/components/Divider';
import { Colors} from 'qualificame-native/app/styles';

import styles from './TextDividerStyle';

export default class TextDivider extends PureComponent {
	render() {
		return (
			<View style={[styles.row, styles.alignItemsCenter, (this.props.addHorizontalMargin && styles.baseHorizontalMargin), (this.props.addVerticalMargin && styles.baseVerticalMargin) ]}>
				<View style={styles.flex1}>
					<Divider color={this.props.color} />
				</View>
				<View style={[styles.alignItemsCenter, styles.smallHorizontalMargin]}>
					<TinyText 
						text={this.props.text}
						color={this.props.color}
					/>
				</View>
				<View style={styles.flex1}>
					<Divider color={this.props.color} />
				</View>
			</View>
		);
	}
}

	TextDivider.propTypes = {
		text: PropTypes.string,
		color: PropTypes.string,
		addHorizontalMargin: PropTypes.bool,
		addVerticalMargin: PropTypes.bool
	}

	TextDivider.defaultProps = {
		text: '',
		color: 'white',
		addHorizontalMargin: false,
		addVerticalMargin: false
	}
