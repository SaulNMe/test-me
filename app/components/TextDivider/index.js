import TextDivider from './TextDivider';

export default TextDivider;
