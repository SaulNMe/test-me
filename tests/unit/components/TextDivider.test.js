import React from 'react';
import { shallow } from 'enzyme';

import TextDivider from 'qualificame-native/app/components/TextDivider';
import Divider from 'qualificame-native/app/components/Divider';
import TinyText from 'qualificame-native/app/components/TinyText';
import { Colors } from 'qualificame-native/app/styles';
import { View } from 'react-native';

function createTestProps (props) {
	return {
		// common props
		text: 'text',
		color: 'blue',
		addHorizontalMargin: true,
		addVerticalMargin: true,
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<TextDivider {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('displays the correct text', () => {
		expect(wrapper.find(TinyText).props().text).toEqual('text');
	});	

	it('displays the correct color text', () => {
		expect(wrapper.find(TinyText).props().color).toEqual('blue');
	});	

	it('displays a vertical margin', () => {
		expect(wrapper.find(View).first().props().style.find(o => o.marginVertical).marginVertical).toBeTruthy();
	});

	it('displays a horizontal margin', () => {
		expect(wrapper.find(View).first().props().style.find(o => o.marginHorizontal).marginHorizontal).toBeTruthy();
	});

	it('displays both Dividers', () => {
		expect(wrapper.find(Divider).getElements().length).toEqual(2);
	});

	describe('with no props', ()=>{
		beforeEach(() => {
			wrapper = createWrapper();
		});
		it('displays no text', () => {
			expect(wrapper.find(TinyText).props().text).toEqual('');
		});	

		it('displays both Dividers', () => {
			expect(wrapper.find(Divider).getElements().length).toEqual(2);
		});

		it('displays no vertical margin', () => {
			expect(wrapper.find(View).first().props().style.find(o => o.marginVertical)).toBeFalsy();
		});

		it('displays no horizontal margin', () => {
			expect(wrapper.find(View).first().props().style.find(o => o.marginHorizontal)).toBeFalsy();
		});

	})
})

// describe('callbacks');
