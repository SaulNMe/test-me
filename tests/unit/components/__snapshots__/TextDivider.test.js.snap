// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component
  style={
    Array [
      Object {
        "flexDirection": "row",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "marginHorizontal": 16,
      },
      Object {
        "marginVertical": 16,
      },
    ]
  }
>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Divider
      color="blue"
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "marginHorizontal": 8,
        },
      ]
    }
  >
    <TinyText
      align=""
      color="blue"
      text="text"
      weight=""
    />
  </Component>
  <Component
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Divider
      color="blue"
    />
  </Component>
</Component>
`;
